### PR TITLE
Refactor code base

### PR DESCRIPTION
**Pull request by zenangst**

### CODE REVIEW

**Comment by RamonGilabert** [comment]
post-10513-code-refactoring-cat-in-bathtu-u295

File: Source/CenterCellCollectionViewFlowLayout.swift [hunk 1]
```diff
@@ -30,11 +28,13 @@ class CenterCellCollectionViewFlowLayout: UICollectionViewFlowLayout {
             }
             
           } else { // First time in the loop
-            candidateAttributes = attributes;
-            continue;
```
Comment (RamonGilabert) [inline Source/CenterCellCollectionViewFlowLayout.swift]: ::o:

File: Source/LightboxController.swift [hunk 2]
```diff
@@ -129,13 +129,8 @@ public class LightboxController: UIViewController {
...
     super.viewDidLoad()
 
-    let width = CGRectGetWidth(view.frame)
-    let height = CGRectGetHeight(view.frame)
-
     collectionSize = view.bounds.size
-    view.addSubview(collectionView)
-    view.addSubview(pageLabel)
-    view.addSubview(closeButton)
+    [collectionView, pageLabel, closeButton].map { self.view.addSubview($0) }
```
Comment (RamonGilabert) [inline Source/LightboxController.swift]: I'm loving this thingy of the mapping! 😄

File: Source/LightboxController.swift [hunk 3]
```diff
@@ -270,9 +267,7 @@ extension LightboxController: UIScrollViewDelegate {
     if !rotating {
       let pageWidth = collectionSize.width
       let currentPage = Int(floor((collectionView.contentOffset.x - pageWidth / 2) / pageWidth) + 1)
```
Comment (RamonGilabert) [inline Source/LightboxController.swift]: We could split those calculations in more lines, what do you think?

File: Source/LightboxDataSource.swift [hunk 4]
```diff
@@ -16,16 +12,14 @@ extension LightboxController: UICollectionViewDataSource {
...
-          imageView: cell.lightboxView.imageView, URL: imageURL) {
-            error in
-            if error == nil {
-              cell.lightboxView.updateViewLayout()
-            }
+    if let imageURL = NSURL(string: image) where config.remoteImages {
+      config.loadImage(
+        imageView: cell.lightboxView.imageView, URL: imageURL) { error in
+          if error == nil {
+            cell.lightboxView.updateViewLayout()
```
Comment (RamonGilabert) [inline Source/LightboxDataSource.swift]: That could be made in a line

File: Source/CenterCellCollectionViewFlowLayout.swift (hunk 1 above)
Comment (richardoti) [inline Source/CenterCellCollectionViewFlowLayout.swift]: Guess, who wrote initial lines?; :smile: 
`It's probably me`

File: Source/LightboxController.swift (hunk 3 above)
Comment (richardoti) [inline Source/LightboxController.swift]: @RamonGilabert agree with you. A good way is to extract this:
`(collectionView.contentOffset.x - pageWidth / 2) / pageWidth`

Comment (zenangst) [inline Source/LightboxController.swift]: Why are you commenting on lines that were not changed? :sunglasses:

File: Source/LightboxController.swift (hunk 2 above)
Comment (zenangst) [inline Source/LightboxController.swift]: Map is the best!

File: Source/LightboxController.swift (hunk 3 above)
Comment (RamonGilabert) [inline Source/LightboxController.swift]: Oh! I was just sayin'...

**Pull request merged**